### ISSUE TITLE
Suspected bug in new ganache-core istanbul version

### ISSUE THREAD

**Issue by ilanDoron** [trufflesuite/ganache-core#510]
Some of our regression tests on kyber repository revert only on new Istanbul version. This is after we did required changes in contracts for Istanbul.

After some investigation the reverting scenario is this.
1. tx1: user approves token to contract A
2. tx2: 
     - contract A calls token.transferFrom(user, contractB, amount)
     - contract B calls token.transfer(user, amount - 1)

if tx is split to two transactions. the revert doesn't happen.

to reproduce the use this repo:
https://github.com/ilanDoron/showGanacheBug/blob/master/README.md

I created a test contract on ropsten and seems this issue isn't there.

## Steps to Reproduce (for bugs)

to reproduce the use this repo, just clone and follow steps to run tests.
https://github.com/ilanDoron/showGanacheBug/blob/master/README.md

## Your Environment

* Ubuntu 16.04 LTS
* node 12.13.0
* npm 6.12.0
* ganache cli - ^6.8.0-istanbul.0
you can find ganache and truffle versions in linked repo.

**Comment by ondratra** [comment]
I've also encountered an error when working with `ERC721` contracts and `Istanbul`. I have adapted `OpenZeppelin`'s `ERC721` implementation to work with Solidity 0.6.0 and Istanbul, but when sending using multiple `transferFrom()` in a single transaction, it fails in **some** situations. 

I cannot provide showcase repository as @ilanDoron did, but I suppose it's the same problem.

My observation when using `ERC721` with Istanbul so far:
Case I) working
- party A sends token 1 to party B
- party A sends token 2 to party B
- party A sends token 3 to party B

Case II) working
- party A sends token 1 to party B
- party B send token 2 to party A

Case III) not working
- party A sends token 1 to party B
- party A sends token 2 to party B
- party B sends token 3 to party A
